tests/src/Mod/Part/App/TopoShapeMapper.cpp

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+#include "gtest/gtest.h"
+#include "src/App/InitApplication.h"
+#include <Mod/Part/App/TopoShape.h>
+#include "Mod/Part/App/TopoShapeMapper.h"
+
+#include <BRepBuilderAPI_MakeEdge.hxx>
+#include <BRepBuilderAPI_MakeShape.hxx>
+#include <BRepPrimAPI_MakeBox.hxx>
+#include <BRepTools_History.hxx>
+#include <BRepTools_ReShape.hxx>
+#include <TopTools_ListOfShape.hxx>
+#include <TopoDS_Edge.hxx>
+
+// NOLINTBEGIN(readability-magic-numbers,cppcoreguidelines-avoid-magic-numbers)
+
+class TopoShapeMapperTest: public ::testing::Test
+{
+protected:
+    static void SetUpTestSuite()
+    {
+        tests::initApplication();
+    }
+
+    void SetUp() override
+    {
+        _docName = App::GetApplication().getUniqueDocumentName("test");
+        App::GetApplication().newDocument(_docName.c_str(), "testUser");
+    }
+
+    void TearDown() override
+    {
+        App::GetApplication().closeDocument(_docName.c_str());
+    }
+
+private:
+    std::string _docName;
+};
+
+TEST_F(TopoShapeMapperTest, shapeHasherTests)
```

Comment:
I personally prefer individual test methods for each major case or set of cases that are being checked. @JohnAD what are your thoughts on that?

Reply:
Yeah, I wavered on this one.   As I was writing it, it was useful to not repeat the setup and keep adding test cases, but it makes sense to split it out some now.

It seems to me that 14 individual test methods would be too many in this case; I'll aim for about 3-4 to keep the cognitive load reasonable.  That make sense, @chennes ?  Maybe a guideline number of assertions?

Reply:
@JohnAD is the testing expert here -- I will go with the flow.